lib/travis/build/script/crystal.rb

```diff
@@ -65,6 +79,29 @@ def validate_version
         end
 
         def linux_latest
+          sh.if "-n $(command -v snap)" do
+            snap_install_crystal '--channel=latest/stable'
+          end
+          sh.else do
+            apt_install_crystal
+          end
+        end
+
+        def linux_nightly
+          sh.if "-n $(command -v snap)" do
+            snap_install_crystal '--channel=latest/edge'
+          end
+          sh.else do
+            sh.failure "Crystal nightlies will only be supported via snap. Use Xenial or later releases."
+          end
+        end
+
+        def snap_install_crystal(options)
+          sh.cmd %Q(sudo apt-get install -y gcc pkg-config git tzdata libpcre3-dev libevent-dev libyaml-dev libgmp-dev libssl-dev libxml2-dev), echo: true
```

Comment:
Might want to consider sending this command's STDOUT and STDERR to `/dev/null` (or `File::NULL`), in order to avoid excess output. It's also OK if you choose to keep it.